pkg/storage/test/tuples.go

```diff
@@ -1215,6 +1217,57 @@ func ReadTest(t *testing.T, datastore storage.OpenFGADatastore) {
 	})
 }
 
+func ReadPageTest(t *testing.T, datastore storage.OpenFGADatastore) {
+	ctx := context.Background()
+	storeID := ulid.Make().String()
+
+	tuplesWritten := []*openfgav1.TupleKey{
+		tuple.NewTupleKey("document:1", "reader", "user:anne"),
+		// read should skip over these
+		tuple.NewTupleKey("document:2", "a", "user:anne"),
+		tuple.NewTupleKey("document:2", "b", "user:anne"),
+		tuple.NewTupleKey("document:2", "c", "user:anne"),
+		tuple.NewTupleKey("document:2", "d", "user:anne"),
+		tuple.NewTupleKey("document:2", "e", "user:anne"),
+		tuple.NewTupleKey("document:2", "f", "user:anne"),
+		tuple.NewTupleKey("document:2", "g", "user:anne"),
+		tuple.NewTupleKey("document:2", "h", "user:anne"),
+		tuple.NewTupleKey("document:2", "j", "user:anne"),
+		// end of skip
+		tuple.NewTupleKey("document:1", "admin", "user:anne"),
+	}
+
+	err := datastore.Write(ctx, storeID, nil, tuplesWritten)
+	require.NoError(t, err)
+
+	t.Run("returns_2_results_when_page_size_2", func(t *testing.T) {
+		tuplesRead, contToken, err := datastore.ReadPage(
+			ctx,
+			storeID,
+			tuple.NewTupleKey("document:1", "", "user:anne"),
+			storage.PaginationOptions{
+				PageSize: 2,
+				From:     "",
+			},
+		)
+		require.NoError(t, err)
+
+		expectedTuples := []*openfgav1.Tuple{
+			{Key: tuple.NewTupleKey("document:1", "admin", "user:anne")},
+			{Key: tuple.NewTupleKey("document:1", "reader", "user:anne")},
+		}
+
+		cmpOpts := []cmp.Option{
+			protocmp.IgnoreFields(protoadapt.MessageV2Of(&openfgav1.Tuple{}), "timestamp"),
+			testutils.TupleCmpTransformer,
+			protocmp.Transform(),
+		}
+		diff := cmp.Diff(expectedTuples, tuplesRead, cmpOpts...)
```

Comment:
Does this warrant a helper? For me, it's fast enough to find usages of `cmp.Diff` and then copy-paste the options 🤷‍♀️

Reply:
We're comparing proto tuples in more than one place, and we have 5 lines of boilerplate every time we have to because of this. It most certainly warrants a helper, for the same reason we have helpers similarly to reduce test boilerplate.